sys/xtimer/xtimer_core.c

```diff
@@ -79,79 +66,64 @@ void xtimer_init(void)
 
     /* register initial overflow tick */
     _lltimer_set(0xFFFFFFFF);
-}
-
-static void _xtimer_now_internal(uint32_t *short_term, uint32_t *long_term)
-{
-    uint32_t before, after, long_value;
-
-    /* loop to cope with possible overflow of _xtimer_now() */
-    do {
-        before = _xtimer_now();
-        long_value = _long_cnt;
-        after = _xtimer_now();
-
-    } while (before > after);
-
-    *short_term = after;
-    *long_term = long_value;
+    _lltimer_ongoing = false;
 }
 
 uint64_t _xtimer_now64(void)
 {
-    uint32_t short_term, long_term;
-
-    _xtimer_now_internal(&short_term, &long_term);
+    /* time sensitive */
+    uint8_t state = irq_disable();
+    _xtimer_now();
+    irq_restore(state);
 
-    return ((uint64_t)long_term << 32) + short_term;
+    return _xtimer_current_time;
```

Comment:
hmm.. returning this global variable here because we know that it is modified from within _xtimer_now() looks a bit dirty. It would be better to have a clean flow of information (i.e. return the value over `_xtimer_now()`, and only update the global where needed).

Reply:
I changed _xtimer_now64 to the fundamental function and _xtimer_now to use the return value of _xtimer_now64. Thanks!